Nightly Sycl compilation errors in TestMathematicalFunctions: error: no matching function for call to 'isnan'

**Describe the bug**
Nightly Sycl build on Ponte Vecchio emits compilation errors with TestMathematicalFunctions.hpp:
```
01:29:52 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_sycl_pv_oneapi_2023_1_0/kokkos/core/unit_test/TestMathematicalFunctions.hpp:1579:9: error: no matching function for call to 'isnan'
01:29:52     if (isnan(static_cast<KE::bhalf_t>(2.f))
01:29:52         ^~~~~
01:29:52 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_sycl_pv_oneapi_2023_1_0/kokkos/core/src/Serial/Kokkos_Serial_Parallel_Range.hpp:82:7: note: in instantiation of member function 'TestIsNaN<Kokkos::Serial>::operator()' requested here
01:29:52       m_functor_reducer.get_functor()(i, update);
01:29:52       ^
01:29:52 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_sycl_pv_oneapi_2023_1_0/kokkos/core/src/Serial/Kokkos_Serial_Parallel_Range.hpp:121:20: note: in instantiation of function template specialization 'Kokkos::Impl::ParallelReduce<Kokkos::Impl::CombinedFunctorReducer<TestIsNaN<Kokkos::Serial>, Kokkos::Impl::FunctorAnalysis<Kokkos::Impl::FunctorPatternInterface::REDUCE, Kokkos::RangePolicy<Kokkos::Serial>, TestIsNaN<Kokkos::Serial>, int>::Reducer>, Kokkos::RangePolicy<Kokkos::Serial>, Kokkos::Serial>::exec<void>' requested here
01:29:52     this->template exec<WorkTag>(update);
01:29:52                    ^
01:29:52 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_sycl_pv_oneapi_2023_1_0/kokkos/core/src/Kokkos_Parallel_Reduce.hpp:1518:13: note: in instantiation of member function 'Kokkos::Impl::ParallelReduce<Kokkos::Impl::CombinedFunctorReducer<TestIsNaN<Kokkos::Serial>, Kokkos::Impl::FunctorAnalysis<Kokkos::Impl::FunctorPatternInterface::REDUCE, Kokkos::RangePolicy<Kokkos::Serial>, TestIsNaN<Kokkos::Serial>, int>::Reducer>, Kokkos::RangePolicy<Kokkos::Serial>, Kokkos::Serial>::execute' requested here
01:29:52     closure.execute();
01:29:52             ^
01:29:52 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_sycl_pv_oneapi_2023_1_0/kokkos/core/src/Kokkos_Parallel_Reduce.hpp:1534:5: note: in instantiation of member function 'Kokkos::Impl::ParallelReduceAdaptor<Kokkos::RangePolicy<Kokkos::Serial>, TestIsNaN<Kokkos::Serial>, int>::execute_impl' requested here
01:29:52     execute_impl(label, policy, functor, return_value);
01:29:52     ^
01:29:52 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_sycl_pv_oneapi_2023_1_0/kokkos/core/src/Kokkos_Parallel_Reduce.hpp:1687:69: note: in instantiation of function template specialization 'Kokkos::Impl::ParallelReduceAdaptor<Kokkos::RangePolicy<Kokkos::Serial>, TestIsNaN<Kokkos::Serial>, int>::execute<int>' requested here
01:29:52   Impl::ParallelReduceAdaptor<PolicyType, FunctorType, ReturnType>::execute(
01:29:52                                                                     ^
01:29:52 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_sycl_pv_oneapi_2023_1_0/kokkos/core/unit_test/TestMathematicalFunctions.hpp:1551:13: note: in instantiation of function template specialization 'Kokkos::parallel_reduce<Kokkos::RangePolicy<Kokkos::Serial>, TestIsNaN<Kokkos::Serial>, int>' requested here
01:29:52     Kokkos::parallel_reduce(Kokkos::RangePolicy<Space>(0, 1), *this, errors);
01:29:52             ^
01:29:52 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_sycl_pv_oneapi_2023_1_0/kokkos/core/unit_test/TestMathematicalFunctions.hpp:1548:17: note: in instantiation of member function 'TestIsNaN<Kokkos::Serial>::run' requested here
01:29:52   TestIsNaN() { run(); }
01:29:52                 ^
01:29:52 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_sycl_pv_oneapi_2023_1_0/kokkos/core/unit_test/TestMathematicalFunctions.hpp:1620:3: note: in instantiation of member function 'TestIsNaN<Kokkos::Serial>::TestIsNaN' requested here
01:29:52   TestIsNaN<TEST_EXECSPACE>();
01:29:52   ^
01:29:52 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_sycl_pv_oneapi_2023_1_0/kokkos/core/src/Kokkos_MathematicalFunctions.hpp:471:34: note: candidate function not viable: no known conversion from 'KE::bhalf_t' (aka 'floating_point_wrapper<sycl::ext::oneapi::bfloat16>') to 'float' for 1st argument
01:29:52 KOKKOS_IMPL_MATH_UNARY_PREDICATE(isnan)
01:29:52                                  ^
01:29:52 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_sycl_pv_oneapi_2023_1_0/kokkos/core/src/Kokkos_MathematicalFunctions.hpp:159:31: note: expanded from macro 'KOKKOS_IMPL_MATH_UNARY_PREDICATE'
01:29:52   KOKKOS_INLINE_FUNCTION bool FUNC(float x) {                                \
01:29:52                               ^
01:29:52 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_sycl_pv_oneapi_2023_1_0/kokkos/core/src/Kokkos_MathematicalFunctions.hpp:471:34: note: candidate function not viable: no known conversion from 'KE::bhalf_t' (aka 'floating_point_wrapper<sycl::ext::oneapi::bfloat16>') to 'double' for 1st argument
01:29:52 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_sycl_pv_oneapi_2023_1_0/kokkos/core/src/Kokkos_MathematicalFunctions.hpp:163:31: note: expanded from macro 'KOKKOS_IMPL_MATH_UNARY_PREDICATE'
01:29:52   KOKKOS_INLINE_FUNCTION bool FUNC(double x) {                               \
01:29:52                               ^
01:29:52 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_sycl_pv_oneapi_2023_1_0/kokkos/core/src/Kokkos_MathematicalFunctions.hpp:471:34: note: candidate function not viable: no known conversion from 'KE::bhalf_t' (aka 'floating_point_wrapper<sycl::ext::oneapi::bfloat16>') to 'long double' for 1st argument
01:29:52 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_sycl_pv_oneapi_2023_1_0/kokkos/core/src/Kokkos_MathematicalFunctions.hpp:167:15: note: expanded from macro 'KOKKOS_IMPL_MATH_UNARY_PREDICATE'
01:29:52   inline bool FUNC(long double x) {                                          \
01:29:52               ^
01:29:52 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_sycl_pv_oneapi_2023_1_0/kokkos/core/src/Kokkos_MathematicalFunctions.hpp:471:34: note: candidate template ignored: requirement 'std::is_integral_v<Kokkos::Experimental::Impl::floating_point_wrapper<sycl::ext::oneapi::bfloat16>>' was not satisfied [with T = KE::bhalf_t]
01:29:52 1 error generated.
01:29:52 make[2]: *** [core/unit_test/CMakeFiles/Kokkos_CoreUnitTest_Serial1.dir/build.make:541: core/unit_test/CMakeFiles/Kokkos_CoreUnitTest_Serial1.dir/serial/TestSerial_MathematicalFunctions2.cpp.o] Error 1
```

This is the list of commits included when compilation errors began:
Drop pre-Kepler logic in Cuda::impl_initialize ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_blake_sycl_pv_oneapi_2023_1_0/12/changes#28061e84dc7c46554e469b00be739465afd29919) / [githubweb](https://github.com/kokkos/kokkos/commit/28061e84dc7c46554e469b00be739465afd29919))
    Enable death tests for fedora rawhide ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_blake_sycl_pv_oneapi_2023_1_0/12/changes#c19926fd511a29c6e903418a925c762b1b010036) / [githubweb](https://github.com/kokkos/kokkos/commit/c19926fd511a29c6e903418a925c762b1b010036))
    core/src: Add half math functions to private header (#6124) ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_blake_sycl_pv_oneapi_2023_1_0/12/changes#1affb05d0b6034539ad5a90bc4ba59a4f655529a) / [githubweb](https://github.com/kokkos/kokkos/commit/1affb05d0b6034539ad5a90bc4ba59a4f655529a))

The bhalf_t occurrences in the error snip indicate this is likely associated with merge of #6124 

**Please also include the following items to support reproducing the bug**
**Reproducer** (Blake PV queue):

SHA: c32f9c90b0e5401cf60bf922cd38daaf831ca8f4

```bash
module load cmake intel-oneapi-compilers/2023.1.0 intel-oneapi-dpl/2022.1.0 git

$KOKKOS_PATH/generate_makefile.bash --with-sycl --arch=INTEL_PVC --compiler=/projects/x86-64-icelake-rocky8/compilers/intel-oneapi-compilers/2023.1.0/gcc/8.5.0/base/6g2jkiv/compiler/2023.1.0/linux/bin-llvm/clang++ --cxxflags="-fp-model=precise" --cmake-flags=-DKokkos_ENABLE_ONEDPL=OFF
```

--- a/core/unit_test/TestMathematicalFunctions.hpp
+++ b/core/unit_test/TestMathematicalFunctions.hpp
@@ -1575,7 +1575,6 @@ struct TestIsNaN {
       ++e;
       KOKKOS_IMPL_DO_NOT_USE_PRINTF("failed isnan(KE::half_t)\n");
     }
-#endif
     if (isnan(static_cast<KE::bhalf_t>(2.f))
 #if !defined(KOKKOS_ENABLE_CUDA)
         || !isnan(quiet_NaN<KE::bhalf_t>::value) ||
@@ -1585,6 +1584,7 @@ struct TestIsNaN {
       ++e;
       KOKKOS_IMPL_DO_NOT_USE_PRINTF("failed isnan(KE::bhalf_t)\n");
     }
+#endif
     if (isnan(3.)
 #ifndef KOKKOS_COMPILER_NVHPC  // FIXME_NVHPC 23.7
         || !isnan(quiet_NaN<double>::value) ||

